diff --git a/browser/app/moz.build b/browser/app/moz.build
index 37f9b0f8bed5d..71210bc367f8d 100644
--- a/browser/app/moz.build
+++ b/browser/app/moz.build
@@ -180,3 +180,19 @@ if CONFIG["MOZ_ASAN"] or CONFIG["MOZ_DEBUG"]:
     WINCONSOLE = True
 else:
     WINCONSOLE = False
+
+if CONFIG["OS_TARGET"] == "WINNT":
+    checkOSBuildFloorp = "windows"
+elif CONFIG["OS_TARGET"] == "Darwin":
+    checkOSBuildFloorp = "mac"
+else:
+    checkOSBuildFloorp = "linux"
+
+for icon in ["document", "document_pdf"]:
+    DEFINES[icon.upper() + "_ICO"] = '"%s/%s/%s/%s/%s.ico"' % (
+        TOPSRCDIR,
+        CONFIG["MOZ_BRANDING_DIRECTORY"],
+        "content",
+        checkOSBuildFloorp,
+        icon,
+    )
