diff --git a/browser/installer/package-manifest.in b/browser/installer/package-manifest.in
index 6ebf2f62c4..f5980b840a 100644
--- a/browser/installer/package-manifest.in
+++ b/browser/installer/package-manifest.in
@@ -195,6 +195,11 @@
 ; [Extensions]
 @RESPATH@/components/extensions-toolkit.manifest
 
+; [Noraneko Files]
+@RESPATH@/chrome/noraneko.manifest
+@RESPATH@/chrome/noraneko@JAREXT@
+@RESPATH@/browser/buildid2
+
 ; Modules
 @RESPATH@/browser/modules/*
 @RESPATH@/modules/*
